Modules/Synchronization/test/Synchronization.Application.Tests/Tests/DomainEvents/ExternalEventCreatedDomainEventTests.cs

```diff
@@ -0,0 +1,24 @@
+﻿using Backbone.DevelopmentKit.Identity.ValueObjects;
+using Backbone.Modules.Synchronization.Domain.DomainEvents.Outgoing;
+using Backbone.Modules.Synchronization.Domain.Entities.Sync;
+using FluentAssertions;
+using Xunit;
+
+namespace Backbone.Modules.Synchronization.Application.Tests.Tests.DomainEvents;
+
+public class ExternalEventCreatedDomainEventTests
+{
+    [Fact]
+    public void Raises_a_domain_event_on_initialization()
+    {
+        var identity = TestDataGenerator.CreateRandomIdentityAddress();
+        var externalEvent = new ExternalEvent(ExternalEventType.IdentityDeletionProcessStatusChanged, IdentityAddress.Parse(identity), 1, new { });
+
+        externalEvent.DomainEvents.Should().HaveCount(1);
+        externalEvent.DomainEvents[0].Should().BeOfType<ExternalEventCreatedDomainEvent>();
+
+        var createdDomainEvent = (ExternalEventCreatedDomainEvent)externalEvent.DomainEvents[0];
+        createdDomainEvent.EventId.Should().Be(externalEvent.Id.Value);
+        createdDomainEvent.Owner.Should().Be(externalEvent.Owner.Value);
```

Comment:
I'd suggest a FluentAssertions extension that allows us to write something like the following:

```cs
var myDomainEvent = entity.Should().HaveSingleDomainEvent<MyDomainEvent>(); 
myDomainEvent.MyProperty.Should().Be("MyValue");
```

The `HaveSingleDomainEvent` method should
* fail if there is more than one domain event AND if the domain event doesn't have the given type
* (in case the assertions succeed) return the domain event with the given type (in this example `MyDomainEvent`)

Please make sure that we get a meaningful error message in case of a failing assertion.

See https://fluentassertions.com/extensibility/ for docs on how to extend FluentAssertions.

This will simplify upcoming tests by a lot.